provider/pkg/await/deployment.go

```diff
@@ -685,23 +685,15 @@ func (dia *deploymentInitAwaiter) checkReplicaSetStatus() {
 	}
 }
 
+// changeTriggeredRollout returns true if the current deployment has a different revision than the last deployment.
+// This is used to determine whether the deployment is rolling out a new revision, which in turn, creates/updates a
+// replica set.
 func (dia *deploymentInitAwaiter) changeTriggeredRollout() bool {
 	if dia.config.lastInputs == nil {
 		return true
 	}
 
-	fields, err := openapi.PropertiesChanged(
-		dia.config.lastInputs.Object, dia.config.currentInputs.Object,
-		[]string{
-			".spec.template.spec",
-		})
-	if err != nil {
-		logger.V(3).Infof("Failed to check whether Pod template for Deployment %q changed",
-			dia.config.currentInputs.GetName())
-		return false
-	}
-
-	return len(fields) > 0
+	return dia.deployment.GetAnnotations()[revision] != dia.config.lastOutputs.GetAnnotations()[revision]
```

Comment:
This is the core logic change, where we determine if a rollout occur based on revision numbers, not spec changes.